### PR TITLE
Test valid upper bounds

### PR DESCRIPTION
**Pull request by Firionus**
### General:

* [x] I followed the guidelines in the Contributing document
* [x] I have checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change
* [x] I have written new tests for my core changes, as applicable
* [x] My submission passes new and existing tests
* [x] I have updated the documentation accordingly

### Type of changes ?
* [x] Bugfix
* [ ] New feature, doesn't change API
* [ ] New feature, adds to API but doesn't change old API
* [ ] New feature, changes old API

### Changes by this PR
- Add VSCode support in .gitignore
- Changed some tests for ArtDmxBuilder, ArtPollReplyBuilder, ArtTimeCodeBuilder. All of the changed tests relate to validation where a number is checked to be in certain bounds.  
It is a good idea to test the lower invalid bound, the lower valid bound, the upper valid bound and the upper invalid bound. This ensures that the bounds are set properly in the implementation.  
This approach fails only if the implementation splits the range into more than three parts (of invalid < valid < invalid). To test this, a property based testing appraoch would be beneficial. I think this is out of scope though and all implementations here only split the range in three parts - so examples should be fine. 
- The new tests revealed 4 bugs when parsing bytes to ints where no offset was applied to ensure non-negative numbers. These bugs were fixed. 

### Current behaviour
Parsing of ArtDmx sequence, ArtDmx physical, ArtPollReply ubeaVersion and ArtPollReply bindIndex results in ints from -128 to 127. 

### New behaviour
Relevant fields parse to ints in range 0 to 255. 

### Other information
*Java noob venting mode on*
Did I just spend a few hours to find and fix bugs that only exist because Java has no unsigned integer data types!? And not even non-`int` literals?!
*Java noob venting mode off*

### CODE REVIEW

File: lib/src/main/java/de/deltaeight/libartnet/builders/ArtDmxBuilder.java [hunk 1]
```diff
@@ -109,9 +109,8 @@ public ArtDmx buildFromBytes(byte[] packetData) {
 
             int subUni = packetData[14];
 
-            return new ArtDmx(packetData[12], packetData[13], packetData[15],
-                    (subUni & 0xFF) >>> 4, subUni & 0xF,
-                    data.clone(), packetData.clone());
+            return new ArtDmx(packetData[12] & 255, packetData[13] & 255, packetData[15], (subUni & 0xFF) >>> 4,
```
Comment (schw4rzlicht) [inline lib/src/main/java/de/deltaeight/libartnet/builders/ArtDmxBuilder.java]: Could you stick to Hex representation for consistency?

File: lib/src/main/java/de/deltaeight/libartnet/builders/ArtPollReplyBuilder.java [hunk 2]
```diff
@@ -282,7 +282,7 @@ public ArtPollReply buildFromBytes(byte[] packetData) {
                     packetData[18],
                     packetData[19],
                     OemCode.getProductByProductCode(packetData[20] << 8 | packetData[21] & 0xFF),
-                    packetData[22],
+                    packetData[22] & 255,
```
Comment (schw4rzlicht) [inline lib/src/main/java/de/deltaeight/libartnet/builders/ArtPollReplyBuilder.java]: Could you stick to Hex representation for consistency?

File: lib/src/main/java/de/deltaeight/libartnet/builders/ArtPollReplyBuilder.java [hunk 3]
```diff
@@ -335,7 +335,7 @@ public ArtPollReply buildFromBytes(byte[] packetData) {
                     EquipmentStyle.getEquipmentStyle(packetData[200]),
                     Arrays.copyOfRange(packetData, 201, 207),
                     Arrays.copyOfRange(packetData, 207, 211),
-                    packetData[211],
+                    packetData[211] & 255,
```
Comment (schw4rzlicht) [inline lib/src/main/java/de/deltaeight/libartnet/builders/ArtPollReplyBuilder.java]: Could you stick to Hex representation for consistency?

**Review by schw4rzlicht — CHANGES_REQUESTED** [review]
Some minor formatting :)

> Did I just spend a few hours to find and fix bugs that only exist because Java has no unsigned integer data types!?

Yes! That's what caused headache for me as well and that's exactly where these bugs came from 🙄 

> And not even non-int literals?!

It has. Unless we have a different understanding of non-int literals 😅

**Comment by Firionus** [comment]
> > And not even non-int literals?!
> 
> It has. Unless we have a different understanding of non-int literals 😅

Well, I meant non-int integer literals. Like http://mail.openjdk.java.net/pipermail/coin-dev/2009-March/000905.html (proposal wasn't succesful).

**Review by schw4rzlicht — APPROVED** [review]

**Comment by schw4rzlicht** [comment]
> Like http://mail.openjdk.java.net/pipermail/coin-dev/2009-March/000905.html 

I missed the context here. Yes, they unfortunately don't exist.

Otherwise this looks good to me, thanks again!

**Comment by schw4rzlicht** [comment]
As per https://status.maven.org I can't deploy right now but I will asap.

**Comment by schw4rzlicht** [comment]
I was finally able to release the changes in `1.1.2-beta`. The artifacts should be available on Maven Central any minute.

**Pull request merged**